### ISSUE TITLE
L016, L039 and L048 errors when using SQLFluff along with dbt

### ISSUE THREAD

**Issue by SoheilSalmani** [sqlfluff/sqlfluff#1434]
Hello, I have the following problem.

## Expected Behaviour

No fail.

## Observed Behaviour

```
❯ sqlfluff lint
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [/workspaces/python/bug_sqlfluff/my_dbt_project/models/example/transactions.sql] FAIL
L:  20 | P:  35 | L016 | Line is too long
L:  20 | P:  35 | L039 | Unnecessary whitespace found.
L:  20 | P:  36 | L016 | Line is too long
L:  20 | P:  36 | L048 | Missing whitespace before '2021-09-2
All Finished 📜 🎉!
```

```
 ❯ sqlfluff fix
==== finding fixable violations ====
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [/workspaces/python/bug_sqlfluff/my_dbt_project/models/example/transactions.sql] FAIL
L:  20 | P:  35 | L039 | Unnecessary whitespace found.
L:  20 | P:  36 | L048 | Missing whitespace before '2021-09-2
==== fixing violations ====
2 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] ...
Attempting fixes...
Persisting Changes...
== [/workspaces/python/bug_sqlfluff/my_dbt_project/models/example/transactions.sql] FAIL
Done. Some operations failed. Please check your files to confirm.
Some errors cannot be fixed or there is another error blocking it.
```

```
❯ sqlfluff parse my_dbt_project/models/example/transactions.sql 
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                newline:                                      '\n'
[L:  2, P:  1]      |                whitespace:                                   '    '
[L:  2, P:  5]      |                select_clause_element:
[L:  2, P:  5]      |                    column_reference:
[L:  2, P:  5]      |                        identifier:                           'transaction_date'
[L:  2, P: 21]      |                comma:                                        ','
[L:  2, P: 22]      |                newline:                                      '\n'
[L:  3, P:  1]      |                whitespace:                                   '    '
[L:  3, P:  5]      |                select_clause_element:
[L:  3, P:  5]      |                    column_reference:
[L:  3, P:  5]      |                        identifier:                           'store_id'
[L:  3, P: 13]      |                comma:                                        ','
[L:  3, P: 14]      |                newline:                                      '\n'
[L:  4, P:  1]      |                whitespace:                                   '    '
[L:  4, P:  5]      |                select_clause_element:
[L:  4, P:  5]      |                    column_reference:
[L:  4, P:  5]      |                        identifier:                           'units_sold'
[L:  4, P: 15]      |            newline:                                          '\n'
[L:  5, P:  1]      |            newline:                                          '\n'
[L:  6, P:  1]      |            [META] dedent:
[L:  6, P:  1]      |            from_clause:
[L:  6, P:  1]      |                keyword:                                      'FROM'
[L:  6, P:  5]      |                whitespace:                                   ' '
[L:  6, P:  6]      |                from_expression:
[L:  6, P:  6]      |                    [META] indent:
[L:  6, P:  6]      |                    from_expression_element:
[L:  6, P:  6]      |                        table_expression:
[L:  6, P:  6]      |                            table_reference:
[L:  6, P:  6]      |                                identifier:                   '`project_id`'
[L:  6, P: 18]      |                                dot:                          '.'
[L:  6, P: 19]      |                                identifier:                   '`dataset_id`'
[L:  6, P: 31]      |                                dot:                          '.'
[L:  6, P: 32]      |                                identifier:                   '`table_id`'
[L:  6, P: 42]      |                    [META] dedent:
[L:  6, P: 42]      |            newline:                                          '\n'
[L:  7, P:  1]      |            newline:                                          '\n'
[L:  8, P:  1]      |            where_clause:
[L:  8, P:  1]      |                keyword:                                      'WHERE'
[L:  8, P:  6]      |                [META] indent:
[L:  8, P:  6]      |                newline:                                      '\n'
[L:  9, P:  1]      |                whitespace:                                   '    '
[L:  9, P:  5]      |                expression:
[L:  9, P:  5]      |                    column_reference:
[L:  9, P:  5]      |                        identifier:                           'units_sold'
[L:  9, P: 15]      |                    whitespace:                               ' '
[L:  9, P: 16]      |                    comparison_operator:                      '>'
[L:  9, P: 17]      |                    whitespace:                               ' '
[L:  9, P: 18]      |                    literal:                                  '0'
[L:  9, P: 19]      |                    [META] placeholder:                       [Type: 'compound', Raw: '{%- if var("env") == \'test\' %}{%- else %}']
[L: 20, P: 35]      |                    newline:                                  '\n'
[L: 10, P: 30]      |                    [META] placeholder:                       [Type: 'block_start', Raw: '{%- else %}']
[L: 20, P: 35]      |                    whitespace:                               '    '
[L: 10, P: 30]      |                    binary_operator:                          'AND'
[L: 10, P: 30]      |                    whitespace:                               ' '
[L: 10, P: 30]      |                    bracketed:
[L: 10, P: 30]      |                        start_bracket:                        '('
[L: 20, P: 35]      |                        [META] indent:
[L: 10, P: 30]      |                        newline:                              '\n'
[L: 10, P: 30]      |                        whitespace:                           '        '
[L: 10, P: 30]      |                        expression:
[L: 10, P: 30]      |                            column_reference:
[L: 10, P: 30]      |                                identifier:                   'transaction_date'
[L: 10, P: 30]      |                            newline:                          '\n'
[L: 10, P: 30]      |                            whitespace:                       '        '
[L: 10, P: 30]      |                            keyword:                          'BETWEEN'
[L: 10, P: 30]      |                            whitespace:                       ' '
[L: 10, P: 30]      |                            literal:                          "'2021-01-01'"
[L: 10, P: 30]      |                            newline:                          '\n'
[L: 10, P: 30]      |                            whitespace:                       '        '
[L: 10, P: 30]      |                            binary_operator:                  'AND'
[L: 10, P: 30]      |                            whitespace:                       ' '
[L: 10, P: 30]      |                            [META] placeholder:               [Type: 'block_start', Raw: '{%- else %}']
[L: 20, P: 36]      |                            literal:                          "'2021-09-23'"
[L: 20, P: 36]      |                        newline:                              '\n'
[L: 20, P: 37]      |                        whitespace:                           '    '
[L: 20, P: 41]      |                        [META] dedent:
[L: 20, P: 41]      |                        end_bracket:                          ')'
[L: 20, P: 42]      |                [META] dedent:
[L: 20, P: 42]      |    [META] placeholder:                                       [Type: 'block_start', Raw: '{%- endif %}']
[L: 22, P:  7]      |    newline:                                                  '\n'
```

## Steps to Reproduce

```bash
pip install sqlfluff==0.6.6 dbt==0.20.2
```

```sql
-- models/example/transactions.sql

SELECT
    transaction_date,
    store_id,
    units_sold

FROM `project_id`.`dataset_id`.`table_id`

WHERE
    units_sold > 0
  {%- if var("env") == 'test' %}
    AND (
        transaction_date
        BETWEEN '{{ var("start_date_test") }}'
        AND '{{ var("execution_date") }}'
    )
  {%- else %}
    AND (
        transaction_date
        BETWEEN '{{ var("start_date_prod") }}'
        AND '{{ var("execution_date") }}'
    )
  {%- endif %}
```

Compiled version generated by dbt:

```sql
--  target/compiled/.../models/example/transactions.sql

SELECT
    transaction_date,
    store_id,
    units_sold

FROM `project_id`.`dataset_id`.`table_id`

WHERE
    units_sold > 0
    AND (
        transaction_date
        BETWEEN '2021-01-01'
        AND '2021-09-23'
    )
```

## Dialect
BigQuery

## Version
sqlfluff, version 0.6.6

## Configuration
```
[sqlfluff]
templater = dbt
dialect = bigquery

[sqlfluff:templater:dbt]
project_dir = my_dbt_project/
profiles_dir = my_dbt_project/secrets/dbt_profiles/
profile = default
```

**Comment by barrywhart** [comment]
I'm getting an error about the variable `env` being undefined. Can you provide more info how to reproduce the problem?

```
== [models/example/transactions.sql] FAIL
L:   0 | P:   0 |  TMP | dbt compilation error on file
                       | 'models/example/transactions.sql', Required var 'env'
                       | not found in config: Vars supplied to transactions = {}
All Finished 📜 🎉!
```
